### PR TITLE
Added Sensor Fusion Tests

### LINKED ISSUE #1204: Make tests for sensor fusion
### Description of the task
Make tests for sensor fusion so we can confident before integrating it

### Acceptance criteria

- [ ] Test new sensor fusion functionality from #1134

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
Added Sensor Fusion Tests for the code added in #1134 

### Testing Done
Added Tests

### Resolved Issues
Resolves #1204 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/sensor_fusion/refbox_data.h [hunk 1]
```diff
@@ -98,91 +97,156 @@ class TeamInfo
...
+          red_cards(red_cards),
+          yellow_card_times(yellow_card_times),
+          yellow_cards(yellow_cards),
+          timeouts(timeouts),
+          timeout_time(timeout_time),
+          goalkeeper(goalkeeper),
+          foul_counter(foul_counter),
+          ball_placement_failures(ball_placement_failures),
+          can_place_ball(can_place_ball),
+          max_allowed_bots(max_allowed_bots)
```
Comment (garethellis0) [inline src/software/sensor_fusion/refbox_data.h]: @MathewMacDougall @akhilveeraghanta - what are your thoughts on moving this to be header-only? I totally see why we might want to here, it's very small, but it's also a bit inconsistent with the rest of the codebase, and I feel that consistency is super important, especially for our team where even slight inconsistencies can really confuse new members.

Comment (MathewMacDougall) [reply src/software/sensor_fusion/refbox_data.h]: Yea I'd vouch for consistency here (so to **not** have it header-only)

**Review by MathewMacDougall — COMMENTED** [review]
Nothing to add beyond Gareth's comments. Glad to have some tests here

File: src/software/sensor_fusion/refbox_data.h (hunk 1 above)
Comment (jonathanlew) [reply src/software/sensor_fusion/refbox_data.h]: refbox_data.h is also header only. My thought is that these classes are basically constructors, getters, and comparators, so they don't "deserve" a source file and in theory could be autogenerated. But anyway, I'll change it for consistency

Comment (jonathanlew) [reply src/software/sensor_fusion/refbox_data.h]: Done

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Review by MathewMacDougall — APPROVED** [review]
LGTM :fire:

**Pull request merged**